src/main/java/pl/joegreen/lambdaFromString/JavaCompilerProvider.java

```diff
@@ -8,27 +8,34 @@
 
 class JavaCompilerProvider {
 
-    static Optional<JavaCompiler> findDefaultJavaCompiler() {
-        Optional<JavaCompiler> eclipseJavaCompiler = getEclipseJavaCompiler();
-        Optional<JavaCompiler> jdkJavaCompiler = getJdkJavaCompiler();
-        if(eclipseJavaCompiler.isPresent()){
-            return eclipseJavaCompiler;
-        }else if(jdkJavaCompiler.isPresent()){
-            return jdkJavaCompiler;
-        }else{
-            return Optional.empty();
-        }
-    }
+	/**
+	 * Obtain (if available) the default JDK compiler or, if running on JRE, the
+	 * Eclipse compiler. Prioritize the JDK compiler since it works for higher Java
```

Comment:
So looks like eclipse compiler stopped on java 8? I'm not sure why changing the priorities matters though, could you maybe elaborate?

Reply:
Yep. I ran the tests both on jdk8 and jdk14. Running on jdk8, both compilers work just fine and all tests execute nicely.
However, on jdk14, the Eclipse compiler bailed out and does not work. Also, the "official" jdk compiler is (in our humble opinion) more likely to support the full feature set, since it is maintained and tested by the Java developers themselves.
This is why we put it first and make the Eclipse compiler the second option and only for up to Java 8.